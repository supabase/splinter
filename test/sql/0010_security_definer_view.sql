begin;
  set local search_path = '';

  -- 0 issues
  select * from lint."0010_security_definer_view";

  create view public.my_view as select 1;

  -- 1 issue
  select * from lint."0010_security_definer_view";

  -- Apply security_invoker
  create or replace view public.my_view with (security_invoker=on) as select 1;

  -- 0 issues
  select * from lint."0010_security_definer_view";


rollback;
