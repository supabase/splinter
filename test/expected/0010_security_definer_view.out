begin;
  set local search_path = '';
  -- 0 issues
  select * from lint."0010_security_definer_view";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create view public.my_view as select 1;
  -- 1 issue
  select * from lint."0010_security_definer_view";
         name          | level |  facing  | categories |                                             description                                             |                   detail                    |                                        remediation                                        |                        metadata                         |              cache_key               
-----------------------+-------+----------+------------+-----------------------------------------------------------------------------------------------------+---------------------------------------------+-------------------------------------------------------------------------------------------+---------------------------------------------------------+--------------------------------------
 security_definer_view | WARN  | EXTERNAL | {SECURITY} | Detects views that are SECURITY DEFINER meaning that they ignore row level security (RLS) policies. | View \`public.my_view\` is SECURITY DEFINER | https://supabase.com/docs/guides/database/database-linter?lint=0010_security_definer_view | {"name": "my_view", "type": "view", "schema": "public"} | security_definer_view_public_my_view
(1 row)

  -- Apply security_invoker
  create or replace view public.my_view with (security_invoker=on) as select 1;
  -- 0 issues
  select * from lint."0010_security_definer_view";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
